tests/unit/test_configuration.py

```diff
@@ -44,15 +61,18 @@ class CustomTestRunner(BaseRunner):
 
 class TestConfiguration(object):
 
-    def test_read_file(self):
-        tn = tempfile.mktemp()
-        tndir = os.path.dirname(tn)
-        with open(tn, "w") as f:
-            f.write(TEST_CONFIG)
-
+    @pytest.mark.parametrize(
+        ("filename", "contents"),
+        list(TEST_CONFIGS)
+    )
+    def test_read_file(self, filename, contents):
+        tndir = tempfile.mkdtemp()
+        file_path = tndir + "/" + filename
```

Comment:
This will not be portable, I fear. Please use `os.path.normpath(os.path.join(...))`. See below.